src/lib.rs

```diff
@@ -343,28 +343,23 @@ impl Config {
         get_variable(package, variable).map_err(|e| e.to_string())
     }
 
-    fn targetted_env_var(&self, var_base: &str) -> Result<String, env::VarError> {
+    fn targetted_env_var(&self, var_base: &str) -> Option<OsString> {
+        // Typically both `TARGET` and `HOST` will be valid utf-8, so we implicitly ignore the
+        // possiblity of them having that failure mode.
         if let Ok(target) = env::var("TARGET") {
-            let host = env::var("HOST")?;
+            let host = env::var("HOST").ok()?;
```

Comment:
Instead of silently returning `None` here if `HOST` happens to not be UTF-8 (and same for `TARGET`), maybe we should make that a panic instead?

Also it seems weird to return `None` if `HOST` is not set at all, it should probably still go into the code below in one way or another.

Both issues were already present before though, so unless you want to improve that let's ignore it for now and put that into an issue so it's not forgotten :)

Reply:
Yeah, I initially had this changed to

```
if let (Ok(target), Ok(host)) = (env::var("TARGET"), env::var("HOST")) { ... }
```

but realised then that it changes the behaviour and figured that maybe there's a reason it written as it is.

But yeah, I think we should panic if `HOST/TARGET` are non-utf8.